web-transport-quinn/src/crypto.rs

```diff
@@ -39,16 +41,5 @@ pub fn sha256(provider: &Provider, cert: &CertificateDer<'_>) -> hash::Output {
         return hash_provider.hash(cert);
     }
 
-    #[cfg(feature = "aws-lc-rs")]
-    {
-        hash::Output::new(aws_lc_rs::digest::digest(&aws_lc_rs::digest::SHA256, cert).as_ref())
-    }
-    #[cfg(all(feature = "ring", not(feature = "aws-lc-rs")))]
-    {
-        return hash::Output::new(ring::digest::digest(&ring::digest::SHA256, cert).as_ref());
-    }
-    #[cfg(not(any(feature = "ring", feature = "aws-lc-rs")))]
-    {
-        panic!("No SHA-256 backend available. Ensure your provider exposes SHA-256 or enable 'ring'/'aws-lc-rs' feature.");
-    }
+    panic!("No SHA-256 backend available. Ensure your provider exposes SHA-256 or enable the 'ring'/'aws-lc-rs' feature.");
```

Comment:
Actually, because of how the `quinn::rustls` module itself is gated behind `rustls-*` provider feature flags in `quinn` (bad), it's impossible to use `web-transport-quinn` without either of these feature flags either.

This is feature flag creep up the dependency chain lol...

Reply:
Anyway this is slightly misleading:

> (Ensure your provider exposes SHA-256) OR (enable the 'ring'/'aws-lc-rs' feature)

If the process-default `CryptoProvider` installed through `CryptoProvider::install_default` doesn't expose a SHA-256 hasher in their cipher suite, then enabling either of the `ring` / `aws-lc-rs` feature on `web-transport-quinn` would change nothing at all. The user would still run into the same panic here...